From 4c335b30eca75b1139912b28954dea132f9205f9 Mon Sep 17 00:00:00 2001
From: hecanyang <hcy@rock-chips.com>
Date: Fri, 10 Jul 2015 10:48:17 +0800
Subject: [PATCH] dts:sdmmc: slow down sdmmc frequence to 37.5MHz for SDK board

SDK board sdmmc can't run 50MHz, otherwise SD initialising will return error.
log as below. slow down sdmmc frequence to 37.5MHz can fix it.

<4>[  802.611901]  [0:    kworker/0:2:  644] rk_sdmmc: BOOT   The card is inserted.  ===!!!!!!==[mmc1]
<6>[  802.838728]  [0:  kworker/u16:0:    6] mmc_host mmc1: Bus speed (slot 0) = 50000000Hz (slot req 400000Hz, actual 3
96825HZ div = 63)
<4>[  802.838858]  [0:  kworker/u16:0:    6] rk_sdmmc: BOOT dw_mci_setup_bus: argue clk_mmc workaround out 800000Hz for
init[mmc1]
<6>[  802.990527]  [0:  kworker/u16:0:    6] mmc_host mmc1: Bus speed (slot 0) = 50000000Hz (slot req 50000000Hz, actual
50000000HZ div = 0)
<4>[  802.990671]  [0:  kworker/u16:0:    6] rk_sdmmc: BOOT dw_mci_setup_bus: argue clk_mmc workaround out normal clock
[mmc1]
<3>[  802.996461]  [0:  kworker/u16:0:    6] mmc1: error -84 whilst initialising SD card

Signed-off-by: hecanyang <hcy@rock-chips.com>
---
 arch/arm64/boot/dts/rk3368-tb_8846.dts | 4 ++--
 arch/arm64/boot/dts/rk3368-tb_mipi.dts | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/rk3368-tb_8846.dts b/arch/arm64/boot/dts/rk3368-tb_8846.dts
index 082692810945..1de26e2b51c8 100644
--- a/arch/arm64/boot/dts/rk3368-tb_8846.dts
+++ b/arch/arm64/boot/dts/rk3368-tb_8846.dts
@@ -282,8 +282,8 @@
 };
 
 &sdmmc {
-		clock-frequency = <50000000>;
-		clock-freq-min-max = <400000 50000000>;
+		clock-frequency = <37500000>;
+		clock-freq-min-max = <400000 37500000>;
 		supports-highspeed;
 		supports-sd;
 		broken-cd;
diff --git a/arch/arm64/boot/dts/rk3368-tb_mipi.dts b/arch/arm64/boot/dts/rk3368-tb_mipi.dts
index e244e4ccbb9a..26a4a32b5b3b 100644
--- a/arch/arm64/boot/dts/rk3368-tb_mipi.dts
+++ b/arch/arm64/boot/dts/rk3368-tb_mipi.dts
@@ -282,8 +282,8 @@
 };
 
 &sdmmc {
-		clock-frequency = <50000000>;
-		clock-freq-min-max = <400000 50000000>;
+		clock-frequency = <37500000>;
+		clock-freq-min-max = <400000 37500000>;
 		supports-highspeed;
 		supports-sd;
 		broken-cd;
-- 
2.35.3

